web/src/narrow.js

```diff
@@ -397,7 +397,6 @@ export function activate(raw_terms, opts) {
                     const adjusted_terms = adjusted_terms_if_moved(raw_terms, target_message);
 
                     if (adjusted_terms === null) {
-                        blueslip.error("adjusted_terms impossibly null");
```

Comment:
For this PR, I believe the error should be removed. Instead, it should be added back into the codebase in PR #28478 because in that PR, we are sending back archived streams.

Reply:
Yeah this state should never occur but since we don't send archived streams to the client but allow searching messages from archived streams, the `adjusted_terms` is `null` for archived stream messages. And clicking on the message from archived stream causes an error.

In the next PR #28478 we send archived streams to the client and there the `adjusted_terms` should never be `null`.